### PR TITLE
Refs #819, Fixes hyphen/dash mismatch in access token.

### PR DESCRIPTION
**Pull request by lukewertz**
See #819.

### CODE REVIEW

**Comment by e0ipso** [comment]
Code looks legit. Can we have test coverage for that?

File: modules/restful_token_auth/src/Plugin/authentication/TokenAuthentication.php [hunk 1]
```diff
@@ -89,6 +89,13 @@ protected function extractToken(RequestInterface $request) {
 
     // Access token may be on the request, or in the headers.
     $input = $request->getParsedInput();
+
+    // If we don't have an $key_name on either the URL or the in the headers,
```
Comment (amitaibu) [inline modules/restful_token_auth/src/Plugin/authentication/TokenAuthentication.php:93]: an $key_name => a $key_name

File: modules/restful_token_auth/src/Plugin/authentication/TokenAuthentication.php [hunk 2]
```diff
@@ -89,6 +89,13 @@ protected function extractToken(RequestInterface $request) {
...
+    // then check again using a hyphen instead of an underscore.
+    if (empty($input[$key_name]) && empty($request->getHeaders()->get($key_name)->getValueString())) {
```
Comment (e0ipso) [inline modules/restful_token_auth/src/Plugin/authentication/TokenAuthentication.php:95]: We don't need the `empty()`.

``` php
if (empty($input[$key_name]) && $request->getHeaders()->get($key_name)->getValueString()) {
```

File: modules/restful_token_auth/src/Plugin/authentication/TokenAuthentication.php (hunk 1 above)
Comment (amitaibu) [inline modules/restful_token_auth/src/Plugin/authentication/TokenAuthentication.php:93]: maybe it's worth also adding in the comment the reason we do this (e.g. Apache's new versions not accepting headers with underscores)

**Comment by e0ipso** [comment]
Merged

**Comment by e0ipso** [comment]
I made some minor modifications before merging it.

**Pull request closed without merging**